connectors/src/connectors/notion/temporal/activities.ts

```diff
@@ -2453,27 +2454,68 @@ export async function upsertDatabaseStructuredDataFromCache({
     return;
   }
 
-  const pageCacheEntries = await NotionConnectorPageCacheEntry.findAll({
+  const pageCacheEntriesCount = await NotionConnectorPageCacheEntry.count({
     where: {
       parentId: databaseId,
       connectorId,
       workflowId: topLevelWorkflowId,
     },
   });
 
-  if (!pageCacheEntries.length) {
+  if (!pageCacheEntriesCount) {
     localLogger.info("No pages found in cache (skipping).");
     return;
   }
 
-  const pagesProperties = pageCacheEntries.map(
-    (p) => JSON.parse(p.pagePropertiesText) as PageObjectProperties
-  );
+  let pagesProperties: PageObjectProperties[] = [];
```

Comment:
I think you could also avoid holding onto this entirely by generating smaller CSVs as you go through the chunks